### DISCUSSION TITLE
Is way to use  nano-staged to run `tsc` ?

### DISCUSSION THREAD

**Discussion by cawa-93** [usmanyunusov/nano-staged#19] [Q&A] [answered]
My project consists of several packages (such as monorepo). Each has its own `tsconfig.json`. I would like to run type checking only in packages that have been modified.

In my case, I don't care which exactly files have been modified:
```json
{
  "packages/package1/**/*.ts": "tsc --noEmit -p packages/package1/tsconfig.json",
  "packages/package2/**/*.ts": "tsc --noEmit -p packages/package2/tsconfig.json",
  "packages/package3/**/*.ts": "tsc --noEmit -p packages/package3/tsconfig.json"
}
```

When I tried I got error:
```
error TS5042: Option 'project' cannot be mixed with source files on a command line.
```

**Comment by usmanyunusov** [comment, marked as answer]
After release `JS API`, it will be possible to use `tsc`.

**Comment by usmanyunusov** [comment]
@cawa-93 Were you able to run type checking only in the changed packages?

**Reply by cawa-93** [reply]
Yeap
